libs/s25main/network/GameServer.cpp

```diff
@@ -430,13 +433,111 @@ void GameServer::Stop()
     LOG.write("server state changed to stop\n");
 }
 
+// Check if there are players that have not been assigned a team but only a
+// range. Those players are assigned a team now while we try to balanace the
+// number of players per team. Returns true iff players have been assigned.
+bool GameServer::assignPlayersOfRandomTeams(std::vector<JoinPlayerInfo>& playerInfos)
+{
+    std::set<unsigned> PotentialPlayers[4];
```

Comment:
I know this isn't introduced by you but could you use a named constant instead of the 4? You use it quite a lot so a constant makes sense. Maybe rename the NUM_TEAMS to NUM_TEAM_OPTIONS (careful to do it in all code) and introduce a new NUM_TEAMS=4